### ISSUE TITLE
[Feature] Support Python Stored procedure as a way to submit python models

### ISSUE THREAD

**Issue by ChenyuLInx** [dbt-labs/dbt-bigquery#1086]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-bigquery functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Big Query support running Spark as stored procedure. [Link](https://cloud.google.com/bigquery/docs/spark-procedures)
We can add a new submission method in dbt-bigquery to run python models leveraging it.

Basically dbt would still do jinja templating as current but when submitting the job, it will create/replace a stored procedure, and call it.

There are also some ways to [get output](https://cloud.google.com/bigquery/docs/spark-procedures#pass-input-output-parameter) from the stored procedure that we might be able to integrate with

**Comment by ChenyuLInx** [comment]
I am trying to follow the tutorial [here](https://cloud.google.com/bigquery/docs/spark-procedures#create-spark-procedure) to create a python stored procedure, and found that I got stuck trying to to create a connection.
On page https://cloud.google.com/bigquery/docs/working-with-connections#bq_1, I am not sure how to create the connection I need to use in stored prodedure.

**Comment by ChenyuLInx** [comment]
Talked to BQ team yesterday(thanks) and got a walk-through of how to give access to the the connection account created by BQ.
Will pick this up soon!

**Comment by velascoluis** [comment]
hi @ChenyuLInx draft some code here https://github.com/velascoluis/dbt-bigquery/commit/a8249c1532be032acd6dbfb3d6768d86ac1923e6 
Can you have a look perhaps?

**Comment by ChenyuLInx** [comment]
@velascoluis the change set looks great! I will test it locally today.
Do you want to add some test to get a PR going or do you want us to take it over?

**Comment by amychen1776** [comment]
@velascoluis Are there any performance improvements from using Python Stored procs from using Data Procs? I'm trying to understand the differences in the experience and why an user would choose one over the other

**Comment by amychen1776** [comment]
Closing this for now due to recent discussions about support BigQuery Dataframes.
